lib/internal/Magento/Framework/Indexer/Test/Unit/MultiDimensionProviderTest.php

```diff
@@ -221,14 +221,14 @@ private function getDimensionProviderMock($dimensions)
             ->disableOriginalClone()
             ->disableArgumentCloning()
             ->disallowMockingUnknownTypes()
-            ->setMethods(['getIterator'])
+            ->onlyMethods(['getIterator'])
             ->getMockForAbstractClass();
 
         $dimensionProviderMock->expects($this->any())
             ->method('getIterator')
             ->willReturnCallback(
                 function () use ($dimensions) {
-                    return \SplFixedArray::fromArray($dimensions);
+                    return new \ArrayIterator($dimensions);
```

Comment:
looks like we can rollback this too

Reply:
Hello @xmav, if we rollback change, we will get an error in Unit Test :

<img width="1607" alt="Screenshot 2021-09-30 at 19 07 46" src="https://user-images.githubusercontent.com/30952336/135493436-ecb8e534-c425-4a04-90a7-f9cb4b7b1693.png">

Thank you.